### PR TITLE
Simple tests for network/server

### PR DESCRIPTION
**Pull request by ilya**
I made a small refactor that I think is necessary to test the port rebinding thing in proper go fashion but I'm not totally sure.

This includes some 'go fmt' changes that pollute the diff, sorry about that.

### CODE REVIEW

File: network/server.go [hunk 1]
```diff
@@ -133,14 +134,21 @@ func (s *Server) listen() {
...
-func (s *Server) Serve(seconds int) {
+func (s *Server) Serve(seconds int) error {
 	go s.handleMessagesForever()
-	go s.listen()
+
+	listenErrChan := make(chan error)
+	go s.listen(listenErrChan)
+
+	listenErr := <-listenErrChan
+	if (listenErr != nil) {
```
Comment (lacker) [inline network/server.go]: In go style, you don't need parentheses for if statements.

**Comment by lacker** [comment]
Just one nitpick. Also plz confirm that by submitting pull requests you hereby release your code contributions into the public domain ;-)

**Comment by lacker** [comment]
This errchan thing is a good idea, I am happy you came up with it ;-) I have a bunch of other ideas for ways to test the server fails and this strategy will probably be useful

**Comment by ilya** [comment]
I think there might be some race condition here. In the course of making other changes I've been running tests a lot and very infrequently this one will hang. Any ideas?

**Comment by ilya** [comment]
Also, I hereby release my code contributions into the public domain.

**Comment by lacker** [comment]
Closing in favor of PR #2

**Pull request closed without merging**